core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/report/edit/ReportEditViewModel.kt

```diff
@@ -0,0 +1,242 @@
+import com.ustadmobile.core.impl.appstate.LoadingUiState
+import com.ustadmobile.core.impl.nav.UstadSavedStateHandle
+import com.ustadmobile.core.util.ext.replace
+import com.ustadmobile.core.viewmodel.UstadEditViewModel
+import com.ustadmobile.core.viewmodel.report.filteredit.ReportFilterEditViewModel
+import com.ustadmobile.door.ext.withDoorTransactionAsync
+import com.ustadmobile.lib.db.entities.Report
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.kodein.di.DI
+
+data class ReportEditUiState(
+    val reportOptions2: ReportOptions2 = ReportOptions2(),
+)
+
+class ReportEditViewModel(
+    di: DI,
+    savedStateHandle: UstadSavedStateHandle,
+    destName: String = DEST_NAME
+) : UstadEditViewModel(di, savedStateHandle, destName) {
+
+    private val _uiState: MutableStateFlow<ReportEditUiState> =
+        MutableStateFlow(ReportEditUiState())
+    val uiState: Flow<ReportEditUiState> = _uiState.asStateFlow()
+
+    init {
+        loadingState = LoadingUiState.INDETERMINATE
+        val title = systemImpl.getString(MR.strings.edit_report)
+
+        _appUiState.update {
+            AppUiState(
+                title = title,
+                hideBottomNavigation = false
+            )
+        }
+        launchIfHasPermission(
+            setLoadingState = true,
+            permissionCheck = { true }
+        ) {
+            async {
+                loadEntity(
+                    serializer = ReportOptions2.serializer(),
+                    onLoadFromDb = { db ->
+                        val report = db.reportDao().findByUid(entityUidArg)
+                        report?.let {
+                            it.reportOptions?.takeIf { options -> options.isNotBlank() }
+                                ?.let { options ->
+                                    Json.decodeFromString(ReportOptions2.serializer(), options)
+                                } ?: ReportOptions2(title = it.reportTitle ?: "")
+                        }
+                    },
+                    makeDefault = {
+                        ReportOptions2()
+                    },
+                    uiUpdate = { loadedReport ->
+                        _uiState.update { prev ->
+                            prev.copy(
+                                reportOptions2 = loadedReport ?: ReportOptions2()
+                            )
+                        }
+                    }
+                )
+
+                launch {
+                    navResultReturner.filteredResultFlowForKey(RESULT_KEY_REPORT)
+                        .collect { result ->
+                            val reportResult = result.result as? Map<*, *> ?: return@collect
```

Comment:
No need to use a map as the result type. ReportFilterEdit should set the result as the ReportFilter3 itself. Then collect it. No need for the map.